docs/site/Interceptors.md

```diff
@@ -0,0 +1,529 @@
+class DummyController {
+  constructor(
+    @inject('my-controller', {asProxyWithInterceptors: true})
+    public readonly myController: MyControllerWithClassLevelInterceptors,
+  ) {}
+}
+ctx.bind('my-controller').toClass(MyControllerWithClassLevelInterceptors);
+ctx.bind('dummy-controller').toClass(DummyController);
+const dummyController = await ctx.get<DummyController>('dummy-controller');
+const msg = await dummyController.myController.greet('John');
+);
+```
+
+Or:
+
+```ts
+const proxy = await ctx.get<MyControllerWithClassLevelInterceptors>(
+  'my-controller',
+  {asProxyWithInterceptors: true},
+const msg = await proxy.greet('John');
+```
+
+### Use `invokeMethod` to apply interceptors
+
+To explicitly invoke a method with interceptors, use `invokeMethod` from
+`@loopback/context`. Please note `invokeMethod` is used internally by
+`RestServer` for controller methods.
+
+```ts
+import {Context, invokeMethod} from '@loopback/context';
+
+const ctx: Context = new Context();
+
+ctx.bind('name').to('John');
+
+// Invoke a static method
+let msg = await invokeMethod(MyController, 'greetStaticWithDI', ctx);
+
+// Invoke an instance method
+const controller = new MyController();
+msg = await invokeMethod(controller, 'greetWithDI', ctx);
+```
+
+## Apply interceptors
+
+Interceptors form a cascading chain of handlers around the target method
+invocation. We can apply interceptors by decorating methods/classes with
+`@intercept`. Please note `@intercept` does **NOT** return a new method wrapping
+the target one. Instead, it adds some metadata instead and such information is
+used by `invokeMethod` or `invokeWithMethodWithInterceptors` functions to
+trigger interceptors around the target method. The original method stays intact.
+Invoking it directly won't apply any interceptors.
+
+### @intercept
+
+Syntax: `@intercept(...interceptorFunctionsOrBindingKeys)`
+
+The `@intercept` decorator adds interceptors to a class or its methods including
+static and instance methods. Two flavors are accepted:
+
+- An interceptor function
+
+  ```ts
+  class MyController {
+    @intercept(log) // Use the `log` function
+    greet(name: string) {
+      return `Hello, ${name}`;
+    }
+  }
+  ```
+
+- A binding key that can be resolved to an interface function
+
+  ```ts
+  class MyController {
+    @intercept('name-validator') // Use the `name-validator` binding
+    async helloWithNameValidation(name: string) {
+      return `Hello, ${name}`;
+    }
+  }
+
+  // Bind options and provider for `NameValidator`
+  ctx.bind('valid-names').to(['John', 'Mary']);
+  ctx.bind('name-validator').toProvider(NameValidator);
+  ```
+
+### Method level interceptors
+
+A **public** static or prototype method on a class can be decorated with
+`@intercept` to attach interceptors to the target method. Please note
+interceptors don't apply to protected or private methods.
+
+#### Static methods
+
+```ts
+class MyControllerWithStaticMethods {
+  // Apply `log` to a static method
```

Comment:
Please rework this example from a single code snippet with code comments into proper text with multiple code snippets.